src/components/Table/types.ts

```diff
@@ -127,4 +127,6 @@ export type InfiniteScroll = {
   onEndReached: (index: number) => void;
   /** The number of pixels from the bottom of the list to eagerly trigger `onEndReached`. The default is 500px. */
   endOffsetPx?: number;
+  /** When true, will append a loading spinner footer to let the user know the next page is still loading */
+  nextPageLoading?: boolean;
```

Comment:
Thought: Wdyt about `onEndReached` being a callback that returns a `Promise<void>`, and we show the spinner until the promise resolves? Then if the callback does an `await fetchMore`, the spinner will auto-show/auto-resolve without needing the `nextPageLoading` prop?

Reply:
ah! that's a very good idea - will give it a go